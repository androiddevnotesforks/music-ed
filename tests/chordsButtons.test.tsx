import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import ChordsButtons from '../components/chordsButtons'

const mockProps = {
  answers: [{
    name: "Answer Name",
    intervals: [1, 2]
  }],
  examObjs: [{
    notes: ["C4"],
    correctAnswer: ["C4"],
    userAnswer: ["C4"]
  }, {
    notes: ["C4"],
    correctAnswer: ["C4"],
    userAnswer: ["C4"]
  }],
  submitAnswer: jest.fn(),
  currentQ: 1
}

describe("ChordsButtons component", () => {
  beforeEach(() => {
    render(<ChordsButtons {...mockProps}/>)
  })
  it("Renders", () => {
    const chordsButtons = screen.getByTestId("chords-btns-div")
    expect(chordsButtons).toBeInTheDocument()
  })
  it("Matches snapshot", () => {
    const chordsButtons = screen.getByTestId("chords-btns-div")
    expect(chordsButtons).toMatchSnapshot()
  })
  it("Each answer has a button", () => {
    let randIndex = Math.floor(Math.random() * mockProps.answers.length)
    const button = screen.getByTestId(`${mockProps.answers[randIndex].name}-button`)
    expect(button).toBeInTheDocument()
    expect(button.textContent).toMatch(mockProps.answers[randIndex].name)
  })
  it("Button calls submitAnswer", () => {
    let randIndex = Math.floor(Math.random() * mockProps.answers.length)
    const button = screen.getByTestId(`${mockProps.answers[randIndex].name}-button`)
    fireEvent.click(button)
    expect(mockProps.submitAnswer).toHaveBeenCalled()
  })
})