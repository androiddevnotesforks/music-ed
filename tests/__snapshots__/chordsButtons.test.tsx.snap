// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChordsButtons component Matches snapshot 1`] = `
<div
  class="chord-btns-div"
  data-testid="chords-btns-div"
>
  <button
    class="answer-btns active:bg-red-500"
    data-testid="Answer Name-button"
  >
    Answer Name
  </button>
</div>
`;
